Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,111 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
+            int size = exactPostTime.Length; // total "size" posts are there.
+            for(int i=0;i<size-1;i++)
+            {
+                for(int j=i+1;j<size;j++)
+                {
+                    if(exactPostTime[i]==exactPostTime[j])
+                    {
+                        if(showPostTime[i]!=showPostTime[j])
+                        {
+                            return "impossible"; // to check if relevant post-time and showposttime are equally corresponding if equal.
+                        }
+                        else
+                            continue;
+                    }
+                }
+            }
+            string[] result = new string[size]; // results of all the posts are stored in a string array.
+            for(int i=0;i<size;i++)
+            {
+                result[i] = finder(exactPostTime[i],showPostTime[i]);
+            }
+            Array.Sort(result); // to know the current time of now.
+            return result[size-1]; // this is the current time.
+        
         }
+         public static string finder(string exactPostTime_i, string showPostTime_i)
+         {
+             string[] s1 = exactPostTime_i.Split(':'); // split the time by : to get the hour, minute and second strings separately
+             string[] s2 = showPostTime_i.Split(' '); // To know the X->value and if its seconds, minutes and hour ago
+             int hh = int.Parse(s1[0]); // parsed hour value.
+             int mm = int.Parse(s1[1]); // Parsed minute value.
+             int ss = int.Parse(s1[2]); // Parsed seconds Value.
+             int ssnow=ss; // To calculate the current time seconds.

```

Comment:
why did not you use int ssnow = int.Parse(s1[2]);?

Reply:
ssnow, mmnow and hhnow are the later times whose calculation will initiate from hh,mm and ss.
So Parsing was already done for hh, mm and ss variables, hence did the same.